packages/rest/src/lib/CDN.ts

```diff
@@ -154,14 +157,39 @@ export class CDN {
 		return this.makeURL(`/discovery-splashes/${guildId}/${splashHash}`, options);
 	}
 
+	/**
+	 * Generates an emoji's URL for an emoji.
+	 *
+	 * @param emojiId - The emoji id
+	 * @param options - Optional options for the emoji
+	 */
+	public emoji(emojiId: string, options?: Readonly<BaseImageURLOptions>): string;
+
 	/**
 	 * Generates an emoji's URL for an emoji.
 	 *
 	 * @param emojiId - The emoji id
 	 * @param extension - The extension of the emoji
+	 * @deprecated This overload is deprecated. Pass an object containing the extension instead.
 	 */
-	public emoji(emojiId: string, extension?: ImageExtension): string {
-		return this.makeURL(`/emojis/${emojiId}`, { extension });
+	// eslint-disable-next-line @typescript-eslint/unified-signatures
+	public emoji(emojiId: string, extension?: ImageExtension): string;
+
+	public emoji(emojiId: string, options?: ImageExtension | Readonly<BaseImageURLOptions>): string {
+		let resolvedOptions;
+
+		if (typeof options === 'string') {
+			if (!deprecationEmittedForEmoji) {
+				deprecationWarning('Passing a string for CDN#emoji() is deprecated. Use an object instead.');
```

Comment:
This isn't right is it? It's specifically for the second param, which this doesn't indicate to me

Reply:
Would "Passing a string for the extension..." be better?

Reply:
I don't think so, cause you still pass a string to the extension, it'd be more along the lines of "Passing an extension as the second parameter" or something (I don't like what I said there even cause it kinda suggests you can't pass an extensions at all)

Reply:
This?

```suggestion
				deprecationWarning(
					'Passing a string for the second parameter of CDN#emoji() is deprecated. Use a sole object for the first parameter instead.',
				);

```

Reply:
Am I misunderstanding something, the first parameter is still the emojiId regardless right?

Reply:
You're right! I've edited the suggestion.

Reply:
Yeah that's probably fine.